src/services/download/http.rs

```diff
@@ -127,21 +135,22 @@ impl HttpDownloader {
         source: Arc<HttpSourceConfig>,
         filetypes: &'static [FileType],
         object_id: ObjectId,
-    ) -> Vec<SourceFileId> {
+    ) -> Vec<ObjectFileSource> {
         super::SourceLocationIter {
             filetypes: filetypes.iter(),
             filters: &source.files.filters,
             object_id: &object_id,
             layout: source.files.layout,
             next: Vec::new(),
         }
-        .map(|loc| SourceFileId::Http(source.clone(), loc))
+        .map(|loc| HttpObjectFileSource::new(source.clone(), loc).into())
         .collect()
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use super::super::locations::SourceLocation;
```

Comment:
```suggestion
    use crate::locations::SourceLocation;
```

There's another instance of this.

Reply:
I don't get this comment, that location doesn't even exist.

Reply:
Good point, I just quickly saw the `super::super::` and got confused. Turns out, the `locations.rs` file is missing completely from this branch. Could you please check it in?